lucene/directory-provider/src/test/java/org/infinispan/hibernate/search/ClusterTestHelper.java

```diff
@@ -80,22 +80,31 @@ public static FullTextSessionBuilder createClusterNode(Set<Class<?>> entityTypes
    }
 
    /**
-    * Wait some time for the cluster to form
+    * delegates {@link #waitMembersCount(FullTextSessionBuilder, Class, int, long, TimeUnit)} with 10s.
     */
    public static void waitMembersCount(FullTextSessionBuilder node, Class<?> entityType, int expectedSize) {
+      waitMembersCount(node, entityType, expectedSize, 10, TimeUnit.SECONDS);
+   }
+
+   /**
+    * Wait some time for the cluster to form
+    *
+    * @param node Node to be checked
+    * @param entityType An entity for check
+    * @param expectedSize Expected size of the cluster
+    * @param timeout Desired timeout
+    * @param timeoutUnit Timeout units
+     */
+   public static void waitMembersCount(FullTextSessionBuilder node, Class<?> entityType, int expectedSize, long timeout, TimeUnit timeoutUnit) {
+      long endTime = System.currentTimeMillis() + TimeUnit.MILLISECONDS.convert(timeout, timeoutUnit);
       int currentSize = 0;
-      int loopCounter = 0;
-      while (currentSize < expectedSize) {
-         try {
-            Thread.sleep(10);
-         } catch (InterruptedException e) {
-            throw new AssertionFailedError(e.getMessage());
+      do {
+         if (System.currentTimeMillis() > endTime) {
+            throw new AssertionError("Timeout when waiting for desired number of nodes. Expected: " + expectedSize + ", got: " + currentSize);
          }
+         Thread.yield();
```

Comment:
Busy wait for 10 seconds sounds pretty bad. It would be better to sleep for 100ms between iterations, or even better to use `TestingUtil.blockUntilViewReceived()`.